pkg/util/util.go

```diff
@@ -135,3 +136,12 @@ func IsSPDKTgtReady(timeout time.Duration) bool {
 	}
 	return false
 }
+
+func IsMountPointReadOnly(mp mount.MountPoint) bool {
+	for _, opt := range mp.Opts {
+		if opt == "ro" {
+			return true
+		}
+	}
+	return false
+}
```

Comment:
We have a very similar stuff in share manager. I would suggest moving the related code to common-libs, so we can reuse that in all places later. Create a ticket for this later.

[/pkg/server/share_manager.go#L240-L254](https://github.com/longhorn/longhorn-share-manager/blob/de8ff1978f5218656a3a76c3cb6d8ea060d52863/pkg/server/share_manager.go#L240-L254)
```go
func (m *ShareManager) hasHealthyVolume() error {
	mountPath := types.GetMountPath(m.volume.Name)
	if err := exec.CommandContext(m.context, "ls", mountPath).Run(); err != nil {
		return fmt.Errorf(UnhealthyErr, mountPath)
	}

	mounter := mount.New("")
	mountPoints, _ := mounter.List()
	for _, mp := range mountPoints {
		if mp.Path == mountPath && isMountPointReadOnly(mp) {
			return fmt.Errorf(ReadOnlyErr, mountPath)
		}
	}
	return nil
}
```

Reply:
https://github.com/longhorn/longhorn/issues/7394